plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/parameterInfo/KotlinHighLevelFunctionParameterInfoHandler.kt

```diff
@@ -191,6 +191,37 @@ abstract class KotlinHighLevelParameterInfoWithCallHandlerBase<TArgumentList : K
                     }
                 }
 
+                // Determine the parameter to be highlighted.
+                val afterTrailingComma = arguments.isNotEmpty() && currentArgumentIndex == arguments.size
+                val highlightParameterIndex = when {
+                    currentArgumentIndex < arguments.size -> argumentToParameterIndex[arguments[currentArgumentIndex]]
+                    afterTrailingComma -> {
+                        // If last argument is for a vararg parameter, then the argument about to be entered at the cursor should also be
+                        // for that same vararg parameter.
+                        val parameterForLastArgument = argumentMapping[arguments.last()]
+                        if (parameterForLastArgument?.isVararg == true) {
+                            parameterToIndex[parameterForLastArgument]
+                        } else {
+                            null
+                        }
+                    }
+                    else -> null
+                }
+
+                var hasTypeMismatchBeforeCurrent = false
+                for ((index, argument) in arguments.withIndex()) {
+                    if (index >= currentArgumentIndex) break
+                    val parameterForArgument = argumentMapping[argument] ?: continue
+                    if (parameterForArgument == setValueParameter) continue
+
+                    val argumentType = argument.getKtType() ?: error("Argument should have a KtType")
+                    val parameterType = parameterForArgument.annotatedType.type
+                    if (argumentType.isNotSubTypeOf(parameterType)) {
+                        hasTypeMismatchBeforeCurrent = true
+                        break
+                    }
+                }
```

Comment:
It's better to extract calculating of `hasTypeMismatchBeforeCurrent` into a function